datasets/persian_ner/persian_ner.py

```diff
@@ -0,0 +1,226 @@
+                    gen_kwargs={"filepath": os.path.join(data_dir, "test_fold1.txt"), "split": "test"},
+                ),
+            ]
+        elif "fold2" in self.config.name:
+            return [
+                datasets.SplitGenerator(
+                    name=datasets.Split.TRAIN,
+                    # These kwargs will be passed to _generate_examples
+                    gen_kwargs={
+                        "filepath": os.path.join(data_dir, "train_fold2.txt"),
+                        "split": "train",
+                    },
+                ),
+                datasets.SplitGenerator(
+                    name=datasets.Split.TEST,
+                    # These kwargs will be passed to _generate_examples
+                    gen_kwargs={"filepath": os.path.join(data_dir, "test_fold2.txt"), "split": "test"},
+                ),
+            ]
+        elif "fold3" in self.config.name:
+            return [
+                datasets.SplitGenerator(
+                    name=datasets.Split.TRAIN,
+                    # These kwargs will be passed to _generate_examples
+                    gen_kwargs={
+                        "filepath": os.path.join(data_dir, "train_fold3.txt"),
+                        "split": "train",
+                    },
+                ),
+                datasets.SplitGenerator(
+                    name=datasets.Split.TEST,
+                    # These kwargs will be passed to _generate_examples
+                    gen_kwargs={"filepath": os.path.join(data_dir, "test_fold3.txt"), "split": "test"},
+                ),
+            ]
+
+    def _generate_examples(self, filepath, split):
+        """ Yields examples. """
+        # TODO: This method will receive as arguments the `gen_kwargs` defined in the previous `_split_generators` method.
+        # It is in charge of opening the given file and yielding (key, example) tuples from the dataset
+        # The key is not important, it's more here for legacy reason (legacy from tfds)
+
+        # filepath = '/Users/karimfoda/Downloads/ArmanPersoNERCorpus/train_fold1.txt'
+
+        # filepaths = ['/Users/karimfoda/Downloads/ArmanPersoNERCorpus/'+i for i in os.listdir('/Users/karimfoda/Downloads/ArmanPersoNERCorpus/') if 'ReadMe' not in i]
+
+        # labels = []
+        # for filepath in filepaths:
+        with open(filepath, "r", encoding="utf-8") as f:
+
+            for id_, line in enumerate(f):
+
+                if len(line.strip().split(" ")) >= 1:
+                    token = line.strip().split(" ")[0]
+                else:
+                    token = ""
+
+                if len(line.strip().split(" ")) >= 2:
+                    label = line.strip().split(" ")[1]
+                else:
+                    label = ""
+
+                # labels.append(label)
+
+                # print(list(set(labels)))
+                yield id_, {
+                    "token": token,
+                    "label": label,
+                }
```

Comment:
I believe there is a more appropriate implementation here, following my comment about features declaration. The script should be able to group tokens and their NER annotation into sequences (representing sentences).

For example:
```suggestion
        with open(filepath, "r", encoding="utf-8") as f:
            id_ = 0
            tokens = []
            ner_labels = []

            for line in f:
                stripped_line = line.strip(" \n")  # strip away whitespaces AND new line characters
                if len(stripped_line) == 0:
                    # If line is empty, it means we reached the end of a sentence.
                    # We can yield the tokens and labels
                    if len(tokens) > 0 and len(ner_labels) > 0:
                        yield id_, {
                            "tokens": tokens,
                            "ner_labels": ner_labels,
                        }
                    else:
                        # Do not yield if tokens or ner_labels is empty
                        # It can be the case if several empty lines are contiguous
                        continue
                    # Then we need to increment the _id and reset the tokens and ner_labels list
                    id_ += 1
                    tokens = []
                    ner_labels = []
                else:
                    token, ner_label = line.split(' ')  # Retrieve token and label
                    tokens.append(token)
                    ner_labels.append(ner_label)
```

Note that this implementation will not yield the last sentence if the file is not terminated by an empty line. The original data files are terminated by empty lines, but dummy data should be crafted with this in mind.